heartbeat/monitors/task.go

```diff
@@ -73,23 +57,42 @@ func newTask(job Job, config taskConfig, monitor *Monitor) (*task, error) {
 	t.processors = processors
 
 	if err != nil {
-		logp.Critical("Could not create client for monitor task %+v", t.monitor)
-		return nil, errors.Wrap(err, "could not create client for monitor task")
+		logp.Critical("Could not create client for monitor configuredJob %+v", t.monitor)
+		return nil, errors.Wrap(err, "could not create client for monitor configuredJob")
 	}
 
 	return t, nil
 }
 
-func (t *task) prepareSchedulerJob(meta common.MapStr, run jobRunner) scheduler.TaskFunc {
+// jobConfig represents fields needed to execute a single job.
+type jobConfig struct {
+	Name     string             `config:"name"`
+	Type     string             `config:"type"`
+	Schedule *schedule.Schedule `config:"schedule" validate:"required"`
+
+	// Fields and tags to add to monitor.
+	EventMetadata common.EventMetadata    `config:",inline"`
+	Processors    processors.PluginConfig `config:"processors"`
+}
+
+// InvalidMonitorProcessorsError is used to indicate situations when processors could not be loaded.
+// This special type is used because these errors are caught and handled gracefully.
+type InvalidMonitorProcessorsError struct{ root error }
```

Comment:
Uhm... Can we shorten the name a little? `ProcessorsError` or `ConfigError` (more generic) for example?

Reply:
Yep, I like the ring of the second one.